src/v/kafka/server/tests/describe_acls_test.cc

```diff
@@ -0,0 +1,168 @@
+          "test-topic",
+          security::pattern_type::literal},
+        security::acl_entry{
+          security::acl_principal{
+            security::principal_type::user, "User:test-user"},
+          security::acl_host::wildcard_host(),
+          security::acl_operation::read,
+          security::acl_permission::allow}});
+
+    create_acl(
+      security::acl_binding{
+        security::resource_pattern{
+          security::resource_type::sr_subject,
+          "test-topic-value",
+          security::pattern_type::literal},
+        security::acl_entry{
+          security::acl_principal{
+            security::principal_type::user, "User:test-user"},
+          security::acl_host::wildcard_host(),
+          security::acl_operation::read,
+          security::acl_permission::allow}});
+
+    auto client = make_kafka_client().get();
+    auto deferred_close = ss::defer([&client] { client.stop().get(); });
+    client.connect().get();
+
+    kafka::describe_acls_request req;
+    req.data.describe_registry_acls = true;
+    req.data.resource_type = 1;
+    req.data.resource_pattern_type = 1;
+    req.data.operation = 1;
+    req.data.permission_type = 1;
+
+    auto resp = client.dispatch(std::move(req), kafka::api_version(2)).get();
+    BOOST_REQUIRE(!resp.data.errored());
+    BOOST_REQUIRE_EQUAL(resp.data.error_code, kafka::error_code::none);
+    BOOST_REQUIRE_EQUAL(resp.data.resources.size(), 1);
+    BOOST_REQUIRE_EQUAL(resp.data.resources[0].type, 100); // 100 is SR_SUBJECT
+    BOOST_REQUIRE_EQUAL(resp.data.resources[0].name, "test-topic-value");
+    BOOST_REQUIRE(resp.data.resources[0].registry_resource);
+}
+
+FIXTURE_TEST(describe_acls_invalid_for_kafka, describe_acls_fixture) {
+    auto client = make_kafka_client().get();
+    auto deferred_close = ss::defer([&client] { client.stop().get(); });
+    client.connect().get();
+
+    kafka::describe_acls_request req;
+    req.data.describe_registry_acls = false;
+    // Request SR_SUBJECT resource type, which is invalid if
+    // describe_registry_acls is false
+    req.data.resource_type = 100;
+    req.data.resource_pattern_type = 1;
+    req.data.operation = 1;
+    req.data.permission_type = 1;
+
+    auto resp = client.dispatch(std::move(req), kafka::api_version(2)).get();
+    BOOST_REQUIRE(resp.data.errored());
+    BOOST_REQUIRE_EQUAL(
+      resp.data.error_code, kafka::error_code::invalid_request);
+}
+
+FIXTURE_TEST(describe_acls_invalid_for_sr, describe_acls_fixture) {
+    auto client = make_kafka_client().get();
+    auto deferred_close = ss::defer([&client] { client.stop().get(); });
+    client.connect().get();
+
+    kafka::describe_acls_request req;
+    req.data.describe_registry_acls = true;
+    // Request TOPIC resource type, which is invalid if describe_registry_acls
+    // is true
+    req.data.resource_type = 2;
+    req.data.resource_pattern_type = 1;
+    req.data.operation = 1;
+    req.data.permission_type = 1;
```

Comment:
Nitpick: It would be nice to name some of these.  Instead of creating new names, perhaps:
```c++
    req.data.resource_type = to_kafka_resource_type(security::resource_type::topic);
    req.data.resource_pattern_type = to_kafka_pattern_type(security::pattern_type::prefixed);
    req.data.operation = to_kafka_operation(security::acl_operation::read);
    req.data.permission_type = to_kafka_permission(security::acl_permission::alow);
```